main.go

```diff
@@ -70,26 +79,18 @@ func checkConfigFile() string {
 	}
 
 	file, err := os.Open(confFilePath)
-	defer file.Close()
 	if err != nil {
 		exitWithError("Error opening config file:", err)
 	}
+	defer file.Close()
```

Comment:
was previously closed before the error was handled, could result in calling .Close() on a nil value